01_host/05_consensus/block_production.adoc

```diff
@@ -143,6 +150,53 @@ is required to produce a block. For the definitions of _Epoch-Randomness_
 (<<defn-epoch-randomness>>) and _VRF_ functions (<<sect-vrf>>).
 ****
 
+[#block-production-secondary-slots]
+===== Secondary Slots
+****
+**Secondary slots** work along side primary slot to ensure consistent block
+production, as described in <<sect-block-production-lottery>>. The secondary
+assignee of a block is determined by calculating a specific value, stem:[i_d],
+which indicates the index in the authority set (<<defn-authority-list>>). The
+corresponding authority in that set has the right to author a secondary block.
+
+[stem]
+++++
+p larr h("Enc"_("SC")(r,s))\
+i_d larr p mod A_l
+++++
+
+where:
+
+* stem:[r] is the Epoch randomness (<<defn-epoch-randomness>>).
+* stem:[s] is the slot number (<<defn-epoch-slot>>).
+* stem:["Enc"_("SC")(...)] encodes its inner value to the corresponding SCALE value.
+* stem:[h(...)] creates a 256-bit Blake2 hash from its inner value.
+* stem:[A_l] is the lengths of the authority list (<<defn-authority-list>>).
+
+To create the necessary VRF outputs, a transcript is construced (<<sect-vrf>>).
+
```

Comment:
Till here this all makes sense. I would ask you to rewrite in the style of the primary slot babe algorithm, i.e. like an algorithm that is run by each node at the beginning of an epoch. That then also matches with what you said above, that the assignment is know at the beginning of the epoch, The inputs are the same for all validators, so everybody gets the same result.

This so far I would accept as the spec for a plain, previously known, random round robin style slot assignment.